OpenRA.Mods.Common/Traits/Buildings/Gate.cs

```diff
@@ -111,13 +111,12 @@ void INotifyBlockingMove.OnNotifyBlockingMove(Actor self, Actor blocking)
 
 		bool CanRemoveBlockage(Actor self, Actor blocking)
 		{
-			return !self.IsDisabled() && BuildComplete && blocking.AppearsFriendlyTo(self);
+			return !IsTraitDisabled && !IsTraitPaused && building.BuildComplete && blocking.AppearsFriendlyTo(self);
 		}
 
-		protected override void AddedToWorld(Actor self)
+		void INotifyAddedToWorld.AddedToWorld(Actor self)
 		{
-			base.AddedToWorld(self);
-			blockedPositions = Info.Tiles(self.Location);
+			blockedPositions = building.Info.Tiles(self.Location);
```

Comment:
This looks like it can just as safely be done in the constructor because it only uses that trait's `Info`.